apisix/plugins/log-rotate.lua

```diff
@@ -251,7 +263,18 @@ local function rotate()
     if now_time < rotate_time then
         -- did not reach the rotate time
         core.log.info("rotate time: ", rotate_time, " now time: ", now_time)
-        return
+
+        local log_file = assert(io_open(default_logs[DEFAULT_ACCESS_LOG_FILENAME].file, "r"))
+        local access_log_file_size = file_size(log_file)
+        log_file:close()
+
+        log_file = assert(io_open(default_logs[DEFAULT_ERROR_LOG_FILENAME].file, "r"))
+        local error_log_file_size = file_size(log_file)
+        log_file:close()
+
+        if max_size <= 0 or access_log_file_size < max_size and error_log_file_size < max_size then
```

Comment:
```suggestion
        if max_size <= 0 or (access_log_file_size < max_size and error_log_file_size < max_size) then
```

Reply:
It seems the error log file will be rotated when the access log is too big?

Reply:
> It seems the error log file will be rotated when the access log is too big?

yes

Reply:
Should we rotated  access log and error log separately by max_size?

Reply:
> Should we rotated access log and error log separately by max_size?

Let's do it.

Reply:
done